### DISCUSSION TITLE
TrinityやRSEMについて

### DISCUSSION THREAD

**Discussion by ymitaka** [nig-sc/nigsc_homepage2#526] [Q&A]
ゲノムがまだわかっていない種のRNA-seqを行なっているのですが、自分のホームディレクトリ内にTrinityやRSEMをインストールして発現量解析をしようと先月諸々のPATHを設定しました。先週の時点ではこれらのPATHは機能し、Singularityの方ではなく、自分のホームディレクトリにあるTrinity等が使えたのですが、今週RSEMを実行しようとしたところ、機能していたはずのPATHが全てリセットされており、設定し直してもログアウトして再度ログインすると、またPATHの設定がリセットされてしまいます。
（例えばwhich Trinityと入力すると、先週までは自分のホームディレクトリにあるTrinityまでのパスがつながっているのが分かりましたが、今週は強制的に/usr/binのパスにつながってしまいます）

Singularity内のTrinityを使ってalign_and_estimate_abundance.plを使おうとするとエラーが出て実行できないので、自分のホームディレクトリにTrinityをインストールしていたのですが……どうしたら解決するでしょうか？

**Comment by sasamori0619** [comment]
遺伝研スーパーコンピューターのサポートセンターの人間ではないのですが、回答させていただきます。

もし行っていないようでしたら、まずはコマンドライン上で`echo $PATH`とすることで$PATHに任意のディレクトリが追加されているかをご確認ください。
情報が不足しているため明確なことは言えませんが、私個人の経験として「PATHのexportを.bashrcに書き込んでいない」「.bash_profile内の.bashrcを読み込む行を誤って削除してしまっている」「PATHの優先度で`/usr/bin`が上位に置かれている」「新規conda仮想環境のPATHを設定し忘れている」あたりが原因として思いつきます。

対処方法として以下の手段を提案します。


### 1. 任意のパスを .bashrc に書き込む

コマンドライン上で`export PATH= ...`のようにした場合、ログアウト時にはその情報は失われます。
しかし .bashrc 内にexport PATHを書き込んだ場合、sshでのログイン時にこの .bashrc が自動的に読み込まれるため任意のパスを$PATHに追加することができます。

またPATHには優先度の概念があり、`echo $PATH` 時に手前に出力される場所の方が高い優先度を持ちます。
もしTrinityの置いてあるディレクトリが$PATHに追加されていたとしても、/usr/bin の方が高い優先度を持っているのならば主訴のような症状が現れます。
この場合、`export PATH="/home/<user name>/<任意のディレクトリ>:$PATH"` のように追加したいディレクトリを$PATHの手前に置くことで、優先度を高く設定することが出来ます。


### 2. 直接パスで指定する

やや強引な方法ですが、`/home/<user name>/miniconda3/bin/Trinity --help` のような形で実行したいスクリプトまでの絶対パスを指定することで、パスによる優先度の影響を回避できます。


また、もしTrinityをcondaで入れているのならばqsub時もしくはqsubのスクリプト内でconda activateを行っているかをご確認ください。



解析がうまく通ることをお祈りしております。

**Reply by ymitaka** [reply]
sasamori0619さん、回答ありがとうございます。

> コマンドライン上でexport PATH= ...のようにした場合、ログアウト時にはその情報は失われます。

ついこのあいだ間まで、コマンドライン上でexport PATH=…と打ってパスを指定すると、ログアウトしてもそのパスは有効なままだったんですが、今週に入って（何もいじっていないのに）急にパスが無効になってしまいました。

> しかし .bashrc 内にexport PATHを書き込んだ場合、sshでのログイン時にこの .bashrc が自動的に読み込まれるため任意のパスを$PATHに追加することができます。

.bash_profileがなぜか空になっていたので、この中にログイン時に.bashrcを読み込むためのコマンドを記入しました。
加えてPATHも追加したら、問題は解決しました。

ありがとうございました！
